editoast/src/views/infra/delimited_area.rs

```diff
@@ -0,0 +1,837 @@
+    fn from(value: Sign) -> Self {
+        let Sign {
+            track,
+            position,
+            direction,
+            ..
+        } = value;
+        DirectedLocation {
+            track,
+            position,
+            direction,
+        }
+    }
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "delimited_area",
+    params(InfraIdParam),
+    request_body = inline(DelimitedAreaResponse),
+    responses(
+        (status = 200, body = inline(DelimitedAreaResponse), description = "The track ranges between a list entries and exits." ),
+    )
+)]
+async fn delimited_area(
+    Extension(auth): AuthenticationExt,
+    State(db_pool): State<DbConnectionPoolV2>,
+    State(app_state): State<AppState>,
+    Path(InfraIdParam { infra_id }): Path<InfraIdParam>,
+    Json(DelimitedAreaForm { entries, exits }): Json<DelimitedAreaForm>,
+) -> Result<Json<DelimitedAreaResponse>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::InfraWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    // Retrieve the infra
+
+    let conn = &mut db_pool.get().await?;
+    let infra_caches = app_state.infra_caches.clone();
+    let infra =
+        Infra::retrieve_or_fail(conn, infra_id, || InfraApiError::NotFound { infra_id }).await?;
+    let infra_cache = InfraCache::get_or_load(conn, &infra_caches, &infra).await?;
+    let graph = Graph::load(&infra_cache);
+
+    // Retrieve the track ranges
+
+    Ok(Json(DelimitedAreaResponse {
+        track_ranges: track_ranges_from_locations(entries, exits, &graph, &infra_cache),
+    }))
+}
+
+fn track_ranges_from_locations(
+    entries: Vec<DirectedLocation>,
+    exits: Vec<DirectedLocation>,
+    graph: &Graph,
+    infra_cache: &InfraCache,
+) -> Vec<DirectionalTrackRange> {
+    let mut track_ranges: Vec<DirectionalTrackRange> = Vec::new();
+    entries.iter().for_each(|entry| {
+        track_ranges.extend(impacted_tracks(
+            entry,
+            exits.iter().collect::<Vec<&DirectedLocation>>(),
+            graph,
+            infra_cache,
+            MAXIMUM_DISTANCE,
+        ))
+    });
+    track_ranges
+}
+
+fn impacted_tracks(
+    entry: &DirectedLocation,
+    exits: Vec<&DirectedLocation>,
+    graph: &Graph,
+    infra_cache: &InfraCache,
+    max_distance: f64,
+) -> Vec<DirectionalTrackRange> {
+    // Map track identifiers to their list of associated exits:
+    let mut tracks_to_exits: HashMap<&Identifier, Vec<&DirectedLocation>> = HashMap::new();
+    exits.into_iter().for_each(|exit| {
+        let track_id = &exit.track;
+        if let Some(track_exits) = tracks_to_exits.get_mut(track_id) {
+            track_exits.push(exit);
+        } else {
+            let track_exits = vec![exit];
+            tracks_to_exits.insert(track_id, track_exits);
+        };
+    });
+    let exits = tracks_to_exits;
```

Comment:
You can use scopes to make steps more evident:

```suggestion
	let exits = {
		let mut tracks_to_exits = HashMap::new();
		for exit in exits {
			tracks_to_exits.entry(track_id).or_default().push(exit);
		}
		tracks_to_exits
	};
```